pkg/pillar/cmd/downloader/status.go

```diff
@@ -0,0 +1,23 @@
+package downloader
+
+import (
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// Status provides a struct that can be called to update download progress
+type Status interface {
+	Progress(uint)
+}
+
+// DownloadStatus practical implementation of Status
+// requires a context and status
+type DownloadStatus struct {
```

Comment:
Can we use a better name? We have the DownloaderContext as well as the DownloderStatus here.. Why do we need this struct? Can't we just pass these two separately where needed? This struct is a bit confusing..

Reply:
This is an example of where using interfaces can simplify our work. 

The previous download funcs, e.g. [doS3()](https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/downloader/downloader.go#L1172), received `status *types.DownloaderStatus` as an arg, and then had to know the structure of it and to publish it. This made the download responsible for publishing, and being tightly tied to the structure.

By replacing that with an `interface` of `Status`, the download need only know it is receiving something that has a `Progress()` call. This completely decouples it. We can test `download()` without having it publish anything; we can connect it to an entirely different system of informing status updates.

The `DownloadStatus` is a practical implementation of `Status interface`, one that has a context and knows how to publish. 

Given that, I think a better name would be `PublishStatus`, since it is a practical implementation of `Status interface` that, well, publishes to `pubsub`. I will do that.